### DISCUSSION TITLE
자동목차에 제목3도 포함할 수 있는 옵션 부탁드려요.

### DISCUSSION THREAD

**Discussion by sniper0673** [pronist/hELLO#178] [Q&A] [answered]
먼저 깔끔한 스킨 너무너무 잘 쓰고 있습니다. 사용한지는 얼마되지 않았는데.. 완전 제 스타일입니다. 

자동목차도 너무 마음에 드는데.. 
제목3까지 자동목차에 사용할 수 있는 옵션이 있었으면 합니다. 

아니면 어떻게 코드를 수정해야 하는지 아시는 분 있으면.. 가르침 부탁드립니다!

**Comment by pronist** [comment, marked as answer]
@HyuntaMansei 안녕하세요.

대체로 제목3 는 다른 웹페이지에서도 처리하지 않기 때문에, 제 스킨에서도  공식적으로 지원할 예정은 없지만, 코드를 짚어드리자면 헤더 지원을 처리하는 부분은 다음과 같으며 여기에 제목3에 해당하는 태그를 추가하시면 됩니다.

https://github.com/tistory-projects/tistory-skin-hELLO/blob/9e1572eb1c2a5a59c9b8841929b295893787b163/views/Main/Post/Permalink/Post/Content.pug#L24

다음과 같이 처리될 수 있습니다.

```js
supportHeadings: '.contents_style > h2, .contents_style > h3, .contents_style > h4',
```

또한 제목3는 제목2보다 조금 더 들여쓰기가 되어있으므로 이 또한 처리해야합니다.

본문 목차
https://github.com/tistory-projects/tistory-skin-hELLO/blob/9e1572eb1c2a5a59c9b8841929b295893787b163/views/Main/Post/Permalink/Post/Content/TOC.pug#L9

스크롤 목차
https://github.com/tistory-projects/tistory-skin-hELLO/blob/9e1572eb1c2a5a59c9b8841929b295893787b163/views/Main/Post/Permalink/Post/Content/Scrollspy.pug#L33

다음과 같은 느낌입니다.

```html
:class='{ "pl-0": heading.tagName === "H2", "pl-4": heading.tagName === "H3", "pl-8": heading.tagName === "H4" }'
```

**Comment by sniper0673** [comment]
빠른 답변 감사합니다~ 
코드 에디팅과.. 컴파일을 어떻게 하는지 몰라서.. 할 수는 없지만.. 능력 있으신 분들은 이 글을 통해서 수정하시는데 도움이 될 수 있을 것 같습니다.

**Reply by pronist** [reply]
`script.js` 와 `skin.html` 에서도 진행하실 수 있습니다.
